tempto-core/src/main/java/io/trino/tempto/internal/initialization/TestInitializationListener.java

```diff
@@ -251,19 +254,23 @@ private void runBeforeWithContextMethods(ITestResult testResult, TestContext tes
 
     private void runAfterWithContextMethods(ITestResult testResult, TestContext testContext)
     {
-        invokeMethodsAnnotatedWith(AfterTestWithContext.class, testResult, testContext);
+        invokeMethodsAnnotatedWith(ImmutableSet.of(AfterMethodWithContext.class, AfterTestWithContext.class), testResult, testContext);
     }
 
-    private void invokeMethodsAnnotatedWith(Class<? extends Annotation> annotationClass, ITestResult testCase, TestContext testContext)
+    private void invokeMethodsAnnotatedWith(Set<Class<? extends Annotation>> annotationClasses, ITestResult testCase, TestContext testContext)
     {
         for (Method declaredMethod : testCase.getTestClass().getRealClass().getMethods()) {
-            if (declaredMethod.getAnnotation(annotationClass) != null) {
-                try {
-                    declaredMethod.invoke(testCase.getInstance(), reflectionInjectorHelper.getMethodArguments(testContext, declaredMethod));
-                }
-                catch (IllegalAccessException | InvocationTargetException e) {
-                    throw new RuntimeException("error invoking methods annotated with " + annotationClass.getName(), e);
+            for (Class<? extends Annotation> annotationClass : annotationClasses) {
+                if (declaredMethod.getAnnotation(annotationClass) != null) {
+                    try {
+                        declaredMethod.invoke(testCase.getInstance(), reflectionInjectorHelper.getMethodArguments(testContext, declaredMethod));
+                    }
+                    catch (IllegalAccessException | InvocationTargetException e) {
+                        throw new RuntimeException("error invoking method annotated with " + annotationClass.getName(), e);
+                    }
                 }
+                // Invoke annotated method once
+                break;
```

Comment:
Does this `break` statement makes `BeforeTestWithContext` unusable ? Say if I have a method annotated with `@BeforeTestWithContext` it won't be executed today - as the loop is exited only we process or try to process`BeforeMethodWithContext` annotations

Reply:
good catch. it was supposed to be inside the `if`.

now that we broke `BeforeTestWithContext`, the best option is to remove it